v0/destinations/recurly/transform.js

```diff
@@ -0,0 +1,222 @@
+  response.body.JSON = payload;
+  return response;
+};
+
+/**
+ * Create an Account Object from identify call.
+ * Steps ::
+ *      1. First generate address Payload
+ *      2. If postal_code is present, convert that to string.
+ *      3. Now construct complete message data to recurly json format.
+ *      4. Recurly accepts code in lower case format. Make sure to do that otherwise the calls will fail.
+ *      5. Next update bill_to, custom_fields and data.acquisition.
+ *      6. Now check if account already exist. If exist then make PUT call to update else POST call to create resource.
+ * @param {*} message
+ * @param {*} category
+ * @param {*} config
+ * @returns
+ */
+const processIdentify = async (message, category, config) => {
+  const { address } = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.ADDRESS.name]
+  );
+  if (address && address.postal_code) {
+    address.postal_code = address.postal_code.toString();
+  }
+  const data = {
+    ...constructPayload(message, MAPPING_CONFIG[category.name]),
+    address
+  };
+  data.code = data.code.toLowerCase();
+  data.bill_to = BILL_TO_SELF;
+  if (data.acquisition) {
+    data.acquisition.channel = "events";
+  }
+  if (message.customFields) {
+    data.custom_fields = createCustomFields(message.customFields);
+  }
+  if (!data) {
+    throw Error(ErrorMessage.FailedToConstructPayload);
+  }
+  const account = await fetchAccount(data.code, config);
+  if (account.isExist) {
+    delete data.acquisition;
+    delete data.code;
+  }
+  return responseBuilderSimple(
+    data,
+    account.httpMethod,
+    account.endPoint,
+    config.apiKey
+  );
+};
+
+/**
+ * Track call is only supported for two of our ecom events as mention in Config Object.
+ * Track call is mapped to line_items in recurly.
+ * PTN:: RS products array maps to items in recurly.
+ *
+ * Steps::
+ *      1. Check if item exist for line_items to be created.
+ *      2. If the item exist make axios call to get its id and then create a line item.
+ *      3. If the item does not exist on recurly then make axios call to create item on recurly and gets its id.
+ *      4. Next create line_items payload with item_id and payload.
+ * @param {*} message
+ * @param {*} config
+ * @returns
+ */
+const processTrack = async (message, config) => {
+  if (!ECOM_EVENTS.includes(message.event)) {
+    throw Error(ErrorMessage.EcomEventNotSupported);
+  }
+  const { currency } = message;
+  const { products } = message;
+  const accountObj = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.IDENTIFY.name]
+  );
+  if (!accountObj.code) {
+    throw Error(ErrorMessage.FailedToTrackWithoutAccount);
+  }
+  const accountresp = await fetchAccount(accountObj.code, config);
+  if (!accountresp.isExist) {
+    throw Error(ErrorMessage.FailedToTrackWithoutAccount);
+  }
+
+  const response = [];
+
+  for (let i = 0; i < products.length; i += 1) {
+    // eslint-disable-next-line no-param-reassign
```

Comment:
Please remove it. If needed add the rules at the top of the file

Reply:
okkk